### ISSUE TITLE
Uncaught Error when viewing in admin

### ISSUE THREAD

**Issue by genox** [RSC-Labs/medusa-store-analytics#5]
Hi,

I added this plugin to an existing install on the latest version of medusajs and get this error message when clicking on Analytics in the Admin UI:

<img width="1168" alt="Screenshot 2024-03-21 at 09 10 49" src="https://github.com/RSC-Labs/medusa-store-analytics/assets/5611407/78eac720-b060-479b-8946-b26d767616ee">

It seems that there are some issues with components, maybe due to the recent medusajs update. The errors are of the kind "Minified React error" - I only had a look at the deployed test version. I can check on localhost if you would like to have more information but I guess you should be able to reproduce easily by updating medusajs to latest.

Thank you and best regards.

**Comment by radoslaw-sz** [comment]
hi @genox , thanks for raising the issue. Unfortunately, I checked it on a clean environment with newest medusa and I do not have this problem. It looks like it can be related to a data in your store which plugin does not handle properly. Because of that, I need some more information from your side.
Could you please firstly check if it works properly with the older versions of a plugin? The newest one is 0.10.0, but maybe you would be able to check 0.9.0, if error is still there, then 0.8.0 etc. 
Thanks to that I will be able to narrow the issue, to which statistic it is related (often next version adds new statistic).

**Comment by bdavisdotcom** [comment]
Greetings. I pulled this down yesterday and installed it in Medusa 1.20.2.  I too am seeing an error, but it is different.
`SELECT "lineItem"."title" AS "title", "lineItem"."thumbnail" AS "thumbnail", "lineItem"."variant_id" AS "variant_id", "variant"."title" AS "variant_title", SUM("returnItem"."quantity") AS "sum" FROM "public"."return_item" "returnItem" LEFT JOIN "public"."line_item" "lineItem" ON "lineItem"."id" = "returnItem"."item_id"  LEFT JOIN "public"."product_variant" "variant" ON  "variant"."id" = "lineItem"."variant_id" AND "variant"."deleted_at" IS NULL  LEFT JOIN "public"."return" "return" ON "return"."id" = "returnItem"."return_id" WHERE "return"."created_at" >= $1 GROUP BY "lineItem"."title", variant_title, thumbnail, "lineItem"."variant_id" ORDER BY sum DESC LIMIT 5`

`| 2024-03-22 16:52:37.337 UTC [550] ERROR:  column reference "thumbnail" is ambiguous at character 610`

I think the GROUP BY needs "lineItem".thumbnail

`  parameters: [ 2024-03-16T04:00:00.000Z ],
medusa-1    |   driverError: error: column reference "thumbnail" is ambiguous
medusa-1    |       at /home/medusa-app/node_modules/pg/lib/client.js:526:17
medusa-1    |       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
medusa-1    |       at async PostgresQueryRunner.query (/home/medusa-app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:184:25)
medusa-1    |       at async SelectQueryBuilder.loadRawResults (/home/medusa-app/node_modules/typeorm/query-builder/SelectQueryBuilder.js:2192:25)
medusa-1    |       at async SelectQueryBuilder.getRawMany (/home/medusa-app/node_modules/typeorm/query-builder/SelectQueryBuilder.js:646:29)
medusa-1    |       at async ProductsAnalyticsService.getTopReturnedVariantsByCount (/home/medusa-app/node_modules/@rsc-labs/medusa-store-analytics/dist/services/productsAnalytics.js:136:41)`

This is also causing an uncaught exception

**Comment by radoslaw-sz** [comment]
thanks @bdavisdotcom for detailed description - I will make a correction.
Do you see the same error message like mentioned in the first comment? I am wondering if it is the root cause for both problems.

**Comment by bdavisdotcom** [comment]
Greetings. I did see the same uncaught error in the admin like the posted image. When I looked in my system log, that's when I saw the real issue.  It likely may be the same underlying cause.

**Comment by radoslaw-sz** [comment]
hi @bdavisdotcom  / @genox  - possible fix landed in 0.10.1. Please kindly check :)
Unfortunately, I was not able to reproduce it locally on with my test data... I do not know why.. So I need your help to check if it is working.

**Comment by bdavisdotcom** [comment]
Thank you! I will test it out.

**Comment by bdavisdotcom** [comment]
The error seems to be gone now. I can view the analytics.  Thank you!

**Comment by radoslaw-sz** [comment]
thanks @bdavisdotcom !
@genox  if you could kindly check it, I will appreciate, I will be able to close this ticket.

**Comment by genox** [comment]
Hi @radoslaw-sz 

I finally had the time to look at this more carefully and while I encountered initially the same error as before, this is how it looks like in dev:

<img width="597" alt="Screenshot 2024-03-24 at 10 54 35" src="https://github.com/RSC-Labs/medusa-store-analytics/assets/5611407/790c2172-a216-4db9-ac10-90cc18af0641">

I proceeded to raze my lock file, delete node_modules and cache and reinstall all dependencies. Lo and behold, some dependency down the tree was probably not updated and only on reinstall and the error went away.

So all in all, the conclusion is that my initial issue was probably only partially affected by the plugin but it was also a version incompatibility due to how medusajs and yarn seem to have a complicated relationship. I should have considered that from the beginning.

Thank you for your effort! The dashboard looks nice and it fills a big void in the medusa admin. :) Have a nice Sunday.
